Introduce hydra framework with backwards compatibility

jackzhxng opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #11162
* #11081
* __->__ #11029
* #11028

Differential Revision: [D75263989](https://our.internmc.facebook.com/intern/diff/D75263989)

jackzhxng committed:
Introduce hydra framework with backwards compatibility

[ghstack-poisoned]

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"




[ghstack-poisoned]

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"




[ghstack-poisoned]

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"




[ghstack-poisoned]

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"




[ghstack-poisoned]

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"




[ghstack-poisoned]

jackzhxng commented:
@jackzhxng has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D75263989).

jackzhxng commented:
@jackzhxng has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D75263989).

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"


Differential Revision: [D75263989](https://our.internmc.facebook.com/intern/diff/D75263989)

[ghstack-poisoned]

jackzhxng commented:
@jackzhxng has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D75263989).

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"


Differential Revision: [D75263989](https://our.internmc.facebook.com/intern/diff/D75263989)

[ghstack-poisoned]

jackzhxng commented:
@jackzhxng has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D75263989).

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"


Differential Revision: [D75263989](https://our.internmc.facebook.com/intern/diff/D75263989)

[ghstack-poisoned]

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"


Differential Revision: [D75263989](https://our.internmc.facebook.com/intern/diff/D75263989)

[ghstack-poisoned]

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"


Differential Revision: [D75263989](https://our.internmc.facebook.com/intern/diff/D75263989)

[ghstack-poisoned]

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"


Differential Revision: [D75263989](https://our.internmc.facebook.com/intern/diff/D75263989)

[ghstack-poisoned]

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"


Differential Revision: [D75263989](https://our.internmc.facebook.com/intern/diff/D75263989)

[ghstack-poisoned]

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"


Differential Revision: [D75263989](https://our.internmc.facebook.com/intern/diff/D75263989)

[ghstack-poisoned]

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"


Differential Revision: [D75263989](https://our.internmc.facebook.com/intern/diff/D75263989)

[ghstack-poisoned]

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"


Differential Revision: [D75263989](https://our.internmc.facebook.com/intern/diff/D75263989)

[ghstack-poisoned]

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"


Differential Revision: [D75263989](https://our.internmc.facebook.com/intern/diff/D75263989)

[ghstack-poisoned]

larryliu0820 commented on examples/models/llama/config/llm_config_utils.py:
```diff
@@ -0,0 +1,22 @@
+from executorch.examples.models.llama.config.llm_config import LlmConfig
+
+
+def convert_args_to_llm_config(args: argparse.Namespace) -> LlmConfig:
```
I think this should be a static method of `LlmConfig`. For example `LlmConfig.from_args()`

larryliu0820 commented on examples/models/llama/config/targets.bzl:
```diff
@@ -13,3 +13,19 @@ def define_common_targets():
     )
+
+    runtime.python_library(
+        name = "llm_config_utils",
```
Again I think this should be merged into `llm_config`.

larryliu0820 commented on examples/models/llama/export_llama_args.py:
```diff
@@ -0,0 +1,21 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```
Why do we have to have `export_llama_args.py` and `export_llama_hydra.py`? Can we merge them into `export_llama.py`?

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"


Differential Revision: [D75263989](https://our.internmc.facebook.com/intern/diff/D75263989)

[ghstack-poisoned]

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"


Differential Revision: [D75263989](https://our.internmc.facebook.com/intern/diff/D75263989)

[ghstack-poisoned]

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"


Differential Revision: [D75263989](https://our.internmc.facebook.com/intern/diff/D75263989)

[ghstack-poisoned]

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"


Differential Revision: [D75263989](https://our.internmc.facebook.com/intern/diff/D75263989)

[ghstack-poisoned]

jackzhxng committed:
Update on "Introduce hydra framework with backwards compatibility"


Differential Revision: [D75263989](https://our.internmc.facebook.com/intern/diff/D75263989)

[ghstack-poisoned]

kimishpatel commented on examples/models/llama/export_llama.py:
```diff
@@ -4,30 +4,50 @@
+        # The import runs the main function of export_llama_hydra with the remaining args
+        # under the Hydra framework.
+        sys.argv = [arg for arg in sys.argv if arg != "--hydra"]
+        print(f"running with {sys.argv}")
```
did you mean to print this?

kimishpatel commented on examples/models/llama/export_llama.py:
```diff
@@ -4,30 +4,50 @@
+        # under the Hydra framework.
+        sys.argv = [arg for arg in sys.argv if arg != "--hydra"]
+        print(f"running with {sys.argv}")
+        runpy.run_module(
```
not just use importlib?
